src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -479,40 +470,49 @@ internal static void ConfigureApplicationInsights(this ILoggingBuilder builder,
+
+                        if (!string.IsNullOrEmpty(eventLogLevel))
                         {
-                            o.DiagnosticsEventListenerLogLevel = level;
+                            if (Enum.TryParse(eventLogLevel, ignoreCase: true, out EventLevel level))
+                            {
+                                o.DiagnosticsEventListenerLogLevel = level;
+                            }
+                            else
+                            {
+                                throw new InvalidEnumArgumentException($"Invalid `{EnvironmentSettingNames.AppInsightsEventListenerLogLevel}`.");
+                            }
                         }
-                        else
+                        if (!string.IsNullOrEmpty(authString))
                         {
-                            throw new InvalidEnumArgumentException($"Invalid `{EnvironmentSettingNames.AppInsightsEventListenerLogLevel}`.");
+                            o.TokenCredentialOptions = TokenCredentialOptions.ParseAuthenticationString(authString);
                         }
-                    }
-                    if (!string.IsNullOrEmpty(authString))
+                    },
+                    t =>
                     {
-                        o.TokenCredentialOptions = TokenCredentialOptions.ParseAuthenticationString(authString);
-                    }
-                }, t =>
-                {
-                    if (t.TelemetryChannel is ServerTelemetryChannel channel)
-                    {
-                        channel.TransmissionStatusEvent += TransmissionStatusHandler.Handler;
-                    }
+                        if (t.TelemetryChannel is ServerTelemetryChannel channel)
+                        {
+                            channel.TransmissionStatusEvent += TransmissionStatusHandler.Handler;
+                        }
 
-                    t.TelemetryProcessorChainBuilder.Use(next => new WorkerTraceFilterTelemetryProcessor(next));
-                    t.TelemetryProcessorChainBuilder.Use(next => new ScriptTelemetryProcessor(next));
-                });
+                        t.TelemetryProcessorChainBuilder.Use(next => new WorkerTraceFilterTelemetryProcessor(next));
+                        t.TelemetryProcessorChainBuilder.Use(next => new ScriptTelemetryProcessor(next));
+                    });
 
                 builder.Services.ConfigureOptions<ApplicationInsightsLoggerOptionsSetup>();
                 builder.Services.AddSingleton<ISdkVersionProvider, FunctionsSdkVersionProvider>();
 
+                // Configure the meter listener, so we publish Meter API based metrics to application insights.
+                builder.Services.AddSingleton<ITelemetryModule, ApplicationInsightsMetricExporter>();
+                builder.Services.Configure<ApplicationInsightsMetricExporterOptions>(o =>
+                {
+                    o.Meters.Add(HostMetrics.FaasMeterName);
```

Comment:
@RohitRanjanMS this will publish the health metrics as well as function invoke duration to app insights. Is that okay to publish func invoke duration? Or do we not want that to customer app insights?

Reply:
In AppInsights mode, we generate a set of metrics related to function invocations. It would be good to filter out the invoke duration metric. It's redundant and may cause confusion.